web/ui/src/features/river-js/RiverValue.tsx

```diff
@@ -1,30 +1,56 @@
-import React, { FC, Fragment, ReactElement } from 'react';
+import React, { FC, Fragment } from 'react';
 
 import { ObjectField, Value, ValueType } from './types';
 
 import styles from './RiverValue.module.css';
 
 export interface RiverValueProps {
   value: Value;
+  name?: string;
+  nthChild?: number;
 }
 
+// TO track the length of the object key value pair
+const DIVISION_LENGTH = 63;
+
 /**
  * RiverValue emits a paragraph which represents a River value.
  */
-export const RiverValue: FC<RiverValueProps> = (props) => {
+export const RiverValue: FC<RiverValueProps> = ({ value, name, nthChild }) => {
   return (
-    <p className={styles.value}>
-      <ValueRenderer value={props.value} indentLevel={0} />
-    </p>
+    <div className={styles.value}>
+      <ValueRenderer value={value} nthChild={nthChild} name={name} />
+    </div>
   );
 };
 
-type valueRendererProps = RiverValueProps & {
-  indentLevel: number;
-};
+const ValueRenderer: FC<RiverValueProps> = ({ value, name, nthChild }) => {
+  const backgroundColor = nthChild && nthChild % 2 === 1 ? '#f4f5f5' : 'white';
+
+  /**
+   * Renderer for the river format for target and label object
+   */
+  const renderGrid = (partition: ObjectField[]) => {
+    // const gridTemplateColumns = '10% 1% 69%';
+
+    let maxLength = 0;
 
-const ValueRenderer: FC<valueRendererProps> = (props) => {
-  const value = props.value;
+    partition.forEach(({ key }) => {
+      return key.length > maxLength ? (maxLength = key.length) : maxLength;
+    });
+
+    const gridTemplateColumns = maxLength < DIVISION_LENGTH ? '10% 1% 89%' : undefined;
+
+    return partition.map(({ key, value }) => {
+      return (
+        <div key={key} className={styles['grid-layout']} style={{ backgroundColor: backgroundColor, gridTemplateColumns }}>
+          <div className={`${styles['grid-item']} ${styles['grid-key']}`}>{key}</div>
+          <div className={styles['grid-item']}>=</div>
+          <ValueRenderer value={value} name={name} />
```

Comment:
Quick explanation on the change here vs the last commit. There won't be a condition renderer anymore. In general, all the objects will follow this layout which will call the `ValueRenderer` recursively in order to deal with any other nested `ValueType` (including colors). `partition` passing in without simplified to detect the type of the value. Set a width on grid column depending on the max length of the key within the group (This way the `=` symbol will be lined up otherwise there will be a big gap in the shorter key group such as `labels`).